lib/SIL/SILConstants.cpp

```diff
@@ -177,6 +201,16 @@ SymbolicValue::cloneInto(ASTContext &astContext) const {
         memObject->getType(), memObject->getValue(), astContext);
     return getAddress(newMemObject, accessPath, astContext);
   }
+  case RK_Array:
+  case RK_ArrayAddress: {
+    CanType elementType;
+    auto elts = getArrayValue(elementType);
+    SmallVector<SymbolicValue, 4> results;
+    results.reserve(elts.size());
+    for (auto elt : elts)
+      results.push_back(elt.cloneInto(astContext));
+    return getArray(results, elementType, astContext);
+  }
   }
```

Comment:
It seems clone is collapsing the representation of RK_Array and RK_ArrayAddress. It ignores the differences. Is this intended?